### ISSUE TITLE
Bug: Gboard, holding the space button doesn't switch after reboot

### ISSUE THREAD

**Issue by virginviolet** [FredHappyface/Android.EweSticker#76]
### Description

On Gboard, if you hold the space button, a "Change keyboard" selection menu pops up (perhaps you need to enable that behavior in Gboard settings if that's something you want). Non-Google keyboards appear on the list, and after you enable EweStictker in system settings, that includes EweSticker. However, each time I restart my phone, the Ewe keyboard disappears from Gboard's keyborad selection list.
I have to disable and re-enable EweSticker in system settings to get it back. Every time my phone is restarted I have to do this.
This does not occur if I hit the "Show Keyboard button" in the navigation bar, which I for some reason have enabled right now. So it seems the issue is specific to Gboard. And Ewe Sticker.

I have four custom keyboards installed, including Ewe, and then there also is the system default keyboard. Only Ewe keyboard is affected.

### Expected Behavior

Restart my phone, go to a text filed, hold down space button, EweSticker appears on the list of keyboards.

### Actual Behavior

EweSticker does not appear on the list of keyboards. Not until I go to system settings and disable and re-enable EweSticker.

## System Information

Operating System (OS): Android
OS Version: 14
Project Version (if applicable): 20240825

**Comment by FredHappyface** [comment]
Pretty sure this is the behaviour for all 3rd party keyboards? I seem to remember they give you a notice basically saying that the behaviour is wonky after restarting. I'll have a play around but if this is core android behaviour then I'm afraid there's nothing I can do :(

**Comment by FredHappyface** [comment]
Yeah just tried this with the Microsoft SwiftKey app and long pressing space will not open the switcher for this either (EweSticker or SwiftKey) on my device (Google Pixel 8, on Android 14). What device are you using? Most articles on this issue just suggest using the regular keyboard switcher 

![image](https://github.com/user-attachments/assets/0ab20d43-11a8-49c8-8414-8ba659830377)

I'm assuming that is not a viable option in your case?

**Comment by virginviolet** [comment]
Thank you for looking in to this.

> Pretty sure this is the behaviour for all 3rd party keyboards?

No. The issue does not occur with [Irregular Expressions](https://github.com/MobileFirstLLC/irregular-expressions) and [Hoplite Polytonic Keyboard](https://github.com/jeremymarch/HopliteKB-Android). And the Samsung keyboards are there too.

> I seem to remember they give you a notice basically saying that the behaviour is wonky after restarting.

I assume what you mean is this message that appears when you enable a third party keyboard from system settings:

> Note: After a reboot, this app can't start until you unlock your phone.

But that's not the issue.

> just tried this with the Microsoft SwiftKey app and long pressing space will not open the switcher for this either (EweSticker or SwiftKey) on my device 

Strange. I cannot find any option to enable or disable this behavior. But you can hold down the language key (globe icon) for the same menu. I have replaced the language button with a dedicated emoji button through Gboard settings (though changing this setting does not disable holding space button to change language).

What's your long-press keyboard delay set to in Gboard settings?

I also tried SwiftKey now, and it does appear in Gboard's language switch menu. And it sticks after reboot.

Also, it's very non-obvious, but if you have a lot of keyboards/languages, you have to scroll down to see all options.

> I'm assuming that is not a viable option in your case?

It _is_ a viable option! I would rather have that bottom row disabled, but, it's alright.

**Comment by FredHappyface** [comment]
Apologies for the releative silence on this issue - had time to take a look today and can confirm this feature has been added! I'll release a new version asap with this change. Thansk so much for your patience on this one :)

**Comment by FredHappyface** [comment]
Fixed in https://github.com/FredHappyface/Android.EweSticker/releases/tag/20250209

Going to close for now. Thanks 😊
